src/pg/db/pool.js

```diff
@@ -0,0 +1,81 @@
+import { Pool } from 'pg';
+import { config as defaultConfig } from './config';
+
+import debug from 'debug';
+const log = (...text) => debug(`graffy:pg:${text.shift()}`)(text);
+
+export class PgDb {
+  constructor(config) {
+    const {
+      connection = defaultConfig.connection,
+      poolOption = defaultConfig.poolOption,
+      handlers = defaultConfig.handlers,
+    } = config;
+
+    this.pool = new Pool({ ...connection, ...poolOption });
+    const { onError } = handlers;
+    this.pool.on('error', onError);
+  }
+
+  async query(query, type = 'query') {
+    const start = Date.now();
+    query.rowMode = 'array';
+    const res = await this.pool.query(query);
+    const duration = Date.now() - start;
+    log(type, `${{ query: query.text, duration, rows: res.rowCount }}`);
+
+    if (type === 'read') {
+      let result = res || [];
+      result = result.rows.flat();
+      // Each row is an array, as there is only one column returned.
+      log('ReadSQL', result);
+      return result;
+    }
+    if (type === 'write') {
+      log('Rows written', res.rowCount);
+      return res.rowCount;
+    }
+
+    return res;
+  }
+
+  async read(query) {
+    return this.query(query, 'read');
+  }
+
+  async write(query) {
+    return this.query(query, 'write');
+  }
```

Comment:
(minor) We can avoid the "type parameter" with a small refactor:

```suggestion
  async query(query) {
    const start = Date.now();
    query.rowMode = 'array';
    const res = await this.pool.query(query);
    const duration = Date.now() - start;
    log(type, `${{ query: query.text, duration, rows: res.rowCount }}`);

    return res;
  }

  async read(query) {
      const result = await this.query(query);
      result = result.rows.flat();
      // Each row is an array, as there is only one column returned.
      log('Read result', result);
      return result;
  }

  async write(query) {
      const result = this.query(query);
      log('Rows written', res.rowCount);
      return res.rowCount;
  }
```

Reply:
this is better, the reason why i used type is the log of original version:
`log(type, `${{ query: query.text, duration, rows: res.rowCount }}`);`
But it is not necessary so it will look like your suggestion!